tools/wpt/browser.py

```diff
@@ -1029,6 +1029,32 @@ def webdriver_supports_browser(self, webdriver_binary, browser_binary, browser_c
         return True
 
 
+class ContentShell(Browser):
+    """Interface for the Chromium content shell.
+    """
+
+    product = "content_shell"
+    requirements = None
+
+    def download(self, dest=None, channel=None, rename=None):
+        raise NotImplementedError
+
+    def install(self, dest=None, channel=None):
+        raise NotImplementedError
+
+    def install_webdriver(self, dest=None, channel=None, browser_binary=None):
+        raise NotImplementedError
+
+    def find_binary(self, venv_path=None, channel=None):
+        return find_executable("content_shell", venv_path)
```

Comment:
Hey @t-troebst quick question (I'm fairly new to the project).

For this find_executable, do you want users to have installed content_shell within the virutalenv path?

For reference, when I ran your fork, I had content_shell already in my PATH (because I have my chromium/src/out/Default in my PATH). I had to change this line to `return find_executable("content_shell")` to get it to run. 

Just wanted clarification on how you expect users to have set this up.

Reply:
Hm yes, I was planning to do the installation in a separate PR since this is already getting rather big. We can get the snapshots from the same place that we get the chrome snapshots (with the name "content-shell.zip" instead of "chrome-[platform].zip"). At that point we would install them into the venv and run from there. However, in the mean time I think it is better to look for the executable in the OS path. I'll change that.

Reply:
Oh, and the way that this will run in the Chromium CI is through the "run_wpt_tests.py" script that accepts a "-t" parameter which sets the content_shell binary explicitly.

Reply:
Thanks for that explanation!

One more question. For those wanting to run locally on windows and mac os, I am reading this [document](https://chromium.googlesource.com/chromium/src/+/HEAD/docs/testing/web_tests_in_content_shell.md#run-web-tests-directly-with-content-shell).

Do we need to do any special case OS matching for `Content Shell.app` for (mac os x)? 

For example: Mac OS
https://github.com/web-platform-tests/wpt/blob/3f6a593433dfad9fe234a616c0dda1bb3deb9625/tools/wpt/browser.py#L953-L957

Reply:
I don't have a Mac device either but I agree that we should probably look for "Content Shell.app/Contents/MacOS/Content Shell". Our goal is to get this to work on Linux in our CI first and then make sure it works on Windows & Mac.